src/controller/Controller.js

```diff
@@ -0,0 +1,127 @@
+const BridgeGame = require('../model/BridgeGame');
+const WinningBridge = require('../model/WinningBridge');
+const InputView = require('../view/InputView');
+const OutputView = require('../view/OutputView');
+const {
+  MESSAGE,
+  LINE_BREAK,
+  RETRY,
+  QUIT,
+  SUCCESS,
+  FAIL,
+} = require('../utils/constants');
+const { close } = require('../utils/utils');
+
+class Controller {
+  constructor() {
+    this.bridgeGame = new BridgeGame();
+    this.winningBridge = new WinningBridge();
+  }
+
+  gameStart() {
+    OutputView.printMessage(MESSAGE.startGame);
+
+    this.inputBridgeSize();
+  }
+
+  inputBridgeSize() {
+    InputView.readBridgeSize(this.validateSize.bind(this));
+  }
+
+  validateSize(size) {
+    try {
+      WinningBridge.validate(size);
+    } catch (error) {
+      OutputView.printError(error);
+      return this.inputBridgeSize();
+    }
+
+    this.makeWinningBridge(size);
+  }
+
+  makeWinningBridge(size) {
+    OutputView.printMessage(LINE_BREAK);
+    this.winningBridge.makeBridge(Number(size));
+
+    this.inputMoving();
+  }
+
+  inputMoving() {
+    InputView.readMoving(this.validateDirection.bind(this));
+  }
+
+  validateDirection(direction) {
+    try {
+      BridgeGame.validateDirection(direction);
+    } catch (error) {
+      OutputView.printError(error);
+      return this.inputMoving();
+    }
+
+    this.move(direction);
+  }
+
+  move(direction) {
+    const CAN_MOVE = this.bridgeGame.canMove(direction, this.winningBridge);
+    this.bridgeGame.move(direction, CAN_MOVE);
+    OutputView.printMap(this.bridgeGame);
+
+    this.checkSuccess(CAN_MOVE, this.isLastStage());
+  }
+
+  isLastStage() {
+    return this.bridgeGame.isLastStage(this.winningBridge);
+  }
+
+  checkSuccess(CAN_MOVE, isLastStage) {
+    if (CAN_MOVE && isLastStage) {
+      const IS_SUCCEEDED = true;
+      return this.printResult(IS_SUCCEEDED);
+    }
+
+    if (CAN_MOVE) return this.inputMoving();
+    if (!CAN_MOVE) return this.inputGameCommand();
+  }
+
+  inputGameCommand() {
+    InputView.readGameCommand(this.validateCommand.bind(this));
+  }
+
+  validateCommand(command) {
+    try {
+      BridgeGame.validateCommand(command);
+    } catch (error) {
+      OutputView.printError(error);
+      return this.inputGameCommand();
+    }
+
+    this.executeCommand(command);
+  }
+
+  executeCommand(command) {
+    if (command === RETRY) this.retry();
+    if (command === QUIT) {
+      const IS_SUCCEEDED = false;
+      this.printResult(IS_SUCCEEDED);
+    }
```

Comment:
```suggestion
  if (command === RETRY) {
    this.retry();
    return;
  }
  
  const IS_SUCCEDED = false;
  this.printResult(IS_SUCCEDED);
```
앞선 `if` 문에서 `return`을 해주면 뒤따르는 `if` 문을 생략할 수 있을 것 같습니다..!

Reply:
네, 그렇네요!
저는 개인적으로 이 상황에서는 if문 두 개를 사용하는 게 가독성이 좋다고 생각하긴 했습니다!!
좋은 의견 감사드려요 ~ 😄